SW/src/svile/sw/SWarena.php

```diff
@@ -521,12 +521,15 @@ private function start()
                 if ($p->getAttributeMap() != null) {//just to be really sure
                     $p->setHealth($this->pg->configs['join.health']);
                     $p->setFood(20);
+                    $p->setGamemode(Player::SURVIVAL);
```

Comment:
No, this function is called when the game starts.
Setting the gamemode to survival here (in this function) is right. but what i asked you is why setGanemode() is inside this `if ($p->getAttributeMap() != null)` if